### DISCUSSION TITLE
Intro theme "triangles" can be potentially played as a beatmap

### DISCUSSION THREAD

**Discussion by nahkd123** [ppy/osu#15484] [Q&A] [answered]
Somehow I'm managed to play "triangles" beatmap (which is supposed to be unplayable) by using the music player in the beatmap select screen. Is this intentional?

## Steps to reproduce
1. Open beatmap select screen
2. Open music player
3. Click on "triangles"
4. Click play

## Video
https://user-images.githubusercontent.com/20136708/140450501-b4efebb1-12f1-4988-bdb6-8a2a4b3e1cbd.mp4

## Logs
[updater.log](https://github.com/ppy/osu/files/7480165/updater.log)
[database.log](https://github.com/ppy/osu/files/7480167/database.log)
[network.log](https://github.com/ppy/osu/files/7480168/network.log)
[performance.log](https://github.com/ppy/osu/files/7480169/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/7480170/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/7480171/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/7480174/performance-input.log)
[performance-update.log](https://github.com/ppy/osu/files/7480175/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/7480176/runtime.log)

**Comment by peppy** [comment]
The only weird part here is that it doesn't show in the carousel when selected.

**Reply by nahkd123** [reply]
https://github.com/ppy/osu/blob/f6c19c95387d3c6d1973b04a1d5a05948266ad3c/osu.Game/Screens/Select/BeatmapCarousel.cs#L702-L705

Maybe because this method returns ``null`` when all beatmaps are marked as hidden?

**Reply by peppy** [reply]
Potentially. The way the carousel is intended to work is that if a beatmap which is filtered away (due to being hidden or otherwise) is selected by some means (multiplayer game, music controller etc.) it should temporary become visible in the carousel while it is selected. This is already tested for other cases, so there must be a specific reason it's behaving differently here.

**Comment by CenTdemeern1** [comment]
I would like to add that it is also possible to delete triangles this way (although it's of course not permanently gone until exit), and that it's also possible for the other intros, although those don't contain any hitobjects.
Video: https://youtu.be/L2etzx51a3Q

**Comment by Gabixel** [comment, marked as answer]
I believe this will be "fixed" (or.. "prevented", I guess) in the next release! Noticed [this PR](https://github.com/ppy/osu/pull/28890), hoping that its content isn't fooling me.
